poly_borders/help_structures.hpp

```diff
@@ -95,6 +95,12 @@ struct Polygon
   Polygon(m2::RectD const & rect, std::vector<MarkedPoint> && points)
     : m_rect(rect), m_points(std::move(points)) {}
 
+  Polygon(Polygon const &) noexcept = default;
```

Comment:
Which code requires these constructors?

Reply:
Slightly reduced error message (brace yourself):
```
xmemory(748): error C2280: 'poly_borders::MarkedPoint::MarkedPoint(const poly_borders::MarkedPoint &)': attempting to reference a deleted function
poly_borders/help_structures.hpp(90): note: compiler has generated 'poly_borders::MarkedPoint::MarkedPoint' here
poly_borders/help_structures.hpp(90): note: 'poly_borders::MarkedPoint::MarkedPoint(const poly_borders::MarkedPoint &)': function was implicitly deleted because a data member invokes a deleted or inaccessible function 'std::unique_ptr<std::mutex>::unique_ptr(const std::unique_ptr<std::mutex> &)'
memory(3451): note: 'std::unique_ptr<std::mutex>::unique_ptr(const std::unique_ptr<std::mutex> &)': function was explicitly deleted
xmemory(748): note: the template instantiation context (the oldest one first) is
poly_borders/help_structures.hpp(112): note: see reference to class template instantiation 'std::vector<poly_borders::MarkedPoint>' being compiled
vector(676): note: while compiling class template member function 'std::vector<poly_borders::MarkedPoint>::vector(const std::vector<poly_borders::MarkedPoint> &)'
poly_borders/help_structures.hpp(115): note: see the first reference to 'std::vector<poly_borders::MarkedPoint>::vector' in 'poly_borders::Polygon::Polygon'
```

Reply:
Shouldn't MarkedPoint copy constructor and operator= be explicitly deleted, as they have an unique_ptr member?

Reply:
Explicitly deleting `MarkedPoint` copy constructor and assignment operator (and adding defaulted move variants) yields pretty much the same error.

Reply:
Which code requires copy constructor and operator= for Polygon class? I see it's moved in the code.

Reply:
I think it's MSVC bug.
https://godbolt.org/z/GcK7WPr87